pkg/runtime/setup.go

```diff
@@ -470,10 +472,14 @@ func (s *setup) install(id strfmt.UUID) (rerr error) {
 		if err := envDef.ApplyFileTransforms(s.path); err != nil {
 			return errs.Wrap(err, "Could not apply env transforms")
 		}
-	} else {
+	} else if s.supportsHardLinks {
 		if err := s.depot.DeployViaLink(id, envDef.InstallDir, s.path); err != nil {
 			return errs.Wrap(err, "Could not deploy artifact via link")
 		}
+	} else {
```

Comment:
I think you can drop this and make line 468 read 

```
if envDef.NeedsTransforms() || !s.supportsHardlinks {
```

The fact that it will run `ApplyFileTransforms` doesn't matter, since there wouldn't be any to apply. Though for robustness it wouldn't hurt to wrap a conditional around that.